specs/ETL.md

```diff
@@ -0,0 +1,102 @@
+        F --> |dataY| G[egress0];
+        F --> |dataY| H[egress1];
+    end
+
+
+    G --> K((Component2));
+    H --> J((Component3));
+
+```
+
+
+#### Egress Handler
+All component types use an `egressHandler` type for routing transit data to actively subscribed downstream ETL components.
+```mermaid
+flowchart TD;
+    Component-->|Transit_Data|A[Egress0];
+    Component-->|Transit_Data|B[Egress1];
+```
+
+#### Ingress Handler
+All component types also use an `ingressHandler` type for ingesting active transit data from upstream ETL components.
+
+
+### UUID
+All components have a UUID that stores critical identification data. Component IDs are used by higher order abstractions to:
+* Represent a component DAG 
+* Understand when component duplicates occur in the system
+
+Component UUID's constitute of both a randomly generated `UUID` and a deterministic `PID`. This is done to ensure uniqueness of each component instance while also ensuring collision based properties so that components can be reused when viable.
+
+### Activity State Channel
+All component's have an internal activity state to represent their current status of operation. Currently, the only supported values are:
+I. _Inactive_ - Refers to a component with no actively running event loop routine.
+II. _Live_ - Refers to a component with an actively running event loop routine.
+III. _Crashed_ - Refers to a component that's suffered some failure and subsequent `eventLoop` routine death. 
+
+**NOTE - State handling policies by managament abstractions has yet to be properly fleshed out**
+
+
+### Pipe
+Pipes are used to perform local arbitrary computations on some input data. Once input data processing has been completed, the output data is then submitted to its respective destination(s). 
+
+#### Attributes
+* An `ActivityState` channel with a pipeline manager
+* Ingress handler that other components can write to
+* `TransformFunc` - A processing function that performs some data translation/transformation on respective inputs
+* An egressHandler that stores dependencies to write to (i,e. Other pipeline components, invariant engine)
+* A specified output data type
+
+### Oracle 
+Oracles are responsible for collecting data from some external third party _(e.g, L1 geth node, L2 rollup node, etc.)_. As of now, oracle's are configurable through the use of a standard `OracleDefinition` interface that allows developers to write arbitrary oracle logic. 
+The following key interface functions are supported/enforced:
+* `ReadRoutine` - Routine used for reading/polling real-time data for some arbitrarily configured data source
+* `BackTestRoutine` - _Optional_ routine used for sequentially backtesting from some starting to ending block heights. 
+
+Unlike other components, Oracle's actually employ _2 go routines_ to safely operate. This is because the definition routines are ran as a seperate go routine with a communication channel to the actual Oracle event loop. This is visualized below:
```

Comment:
```suggestion
Unlike other components, `Oracles` actually employ _2 go routines_ to safely operate. This is because the definition routines are run as a separate go routine with a communication channel to the actual `Oracle` event loop. This is visualized below:
```